### PR TITLE
Preliminary result migration

### PR DESCRIPTION
**Pull request by anticorrelator**
Adds a preliminary migration in preparation for the results refactor in https://github.com/PrefectHQ/prefect/pull/8164.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/orion/database/migrations/versions/postgresql/2023_01_25_132056_9b53afa4c332_rename_data_column.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            UPDATE flow_run_state
+            SET _data = data
+            WHERE flow_run_state.id in (SELECT id FROM flow_run_state WHERE (has_data IS TRUE) AND (_data IS NULL) LIMIT {batch_size});
+        """
+
+    def move_task_data_in_batches(batch_size):
+        return f"""
+            UPDATE task_run_state
+            SET _data = data
+            WHERE task_run_state.id in (SELECT id FROM task_run_state WHERE (has_data IS TRUE) AND (_data IS NULL) LIMIT {batch_size});
```
Comment (zangell44) [inline src/prefect/orion/database/migrations/versions/postgresql/2023_01_25_132056_9b53afa4c332_rename_data_column.py:60]: > AND (_data IS NULL)

Filtering on `_data` doesn't have an index to use here. Maybe we can iterate over `id` instead?

File: src/prefect/orion/schemas/states.py [hunk 2]
```diff
@@ -69,6 +69,13 @@ class Config:
             "Content must be storable as JSON."
         ),
     )
+    _data: Optional[Any] = Field(
```
Comment (zangell44) [inline src/prefect/orion/schemas/states.py:72]: Hmm do we need this in the api schema? I thought `_data` shouldn't be present in responses

**Review by zangell44 — APPROVED** [review]
A few comments but high level looks good!

**Comment by madkinsz** [comment]
@anticorrelator did you mean to link to a different PR in the summary?

**Comment by anticorrelator** [comment]
> @anticorrelator did you mean to link to a different PR in the summary?



> @anticorrelator did you mean to link to a different PR in the summary?

whoops, yeah not that one

**Comment by anticorrelator** [comment]
closing as we have a better solution to migrate `data` smoothly

**Pull request closed without merging**